sdks/java/core/src/test/java/org/apache/beam/sdk/PipelineTest.java

```diff
@@ -384,6 +384,32 @@ public boolean matches(AppliedPTransform<?, ?, ?> application) {
                 new UnboundedCountingInputOverride())));
   }
 
+  /**
+   * Test {@link Pipeline#visualizePipeline()}.
+   */
+  @Test
+  public void testVisualizePipeline(){
+    PTransform<PCollection<? extends String>, PCollection<String>> myTransform =
+        addSuffix("+");
+
+    PCollection<String> input = pipeline.apply(Create.<String>of(ImmutableList.of("a", "b")));
+
+    PCollection<String> left = input.apply("Left1", myTransform).apply("Left2", myTransform);
+    PCollection<String> right = input.apply("Right", myTransform);
+
+    PCollection<String> both = PCollectionList.of(left).and(right)
+        .apply(Flatten.<String>pCollections());
+
+    String visualizedFlow = pipeline.visualizePipeline();
+    Assert.assertEquals(
+        "[end] Flatten.PCollections\n"
```

Comment:
also confuse to me the proper way to check the return, string comparison is odd. Any suggestion?